docs/workflow/requirements/linux-requirements.md

```diff
@@ -1,65 +1,100 @@
 
 Install the following packages for the toolchain:
 
-- cmake
-- llvm
-- clang
-- build-essential
-- python
-- curl
-- git
-- lldb
-- liblldb-dev
-- libunwind8
-- libunwind8-dev
-- gettext
-- libicu-dev
-- liblttng-ust-dev
-- libssl-dev
-- libkrb5-dev
-- libnuma-dev (optional, enables numa support)
-- zlib1g-dev
-- ninja-build (optional, enables building native code with ninja instead of make)
-
-```
+* CMake
+* llvm
+* clang
+* build-essential
+* python-is-python3
+* curl
+* git
+* lldb
+* liblldb-dev
+* libunwind8
+* libunwind8-dev
+* gettext
+* libicu-dev
+* liblttng-ust-dev
+* libssl-dev
+* libkrb5-dev
+* libnuma-dev (optional, enables numa support)
+* zlib1g-dev
+* ninja-build (optional, enables building native code with ninja instead of make)
+
+```bash
 sudo apt-get install -y cmake llvm clang \
-build-essential python curl git lldb liblldb-dev \
+build-essential python-is-python3 curl git lldb liblldb-dev \
 libunwind8 libunwind8-dev gettext libicu-dev liblttng-ust-dev \
 libssl-dev libnuma-dev libkrb5-dev zlib1g-dev ninja-build
 ```
 
 You now have all the required components.
-*Unsupported OSes*:
-In case you have Gentoo you can run following commands:
 
+#### Additional Requirements for Cross-Building
+
+If you are planning to use your Linux environment to do cross-building for other architectures (e.g. Arm32, Arm64) and/or other operating systems (e.g. Alpine, FreeBSD), you need to install these additional dependencies:
+
+* qemu
+* qemu-user-static
+* binfmt-support
+* debootstrap
+* binutils-arm-linux-gnueabihf _(for Arm32 cross-building)_
+* binutils-aarch64-linux-gnu   _(for Arm64 cross-building)_
+* binutils-arm-linux-gnueabi   _(for Armel cross-building)_
+
+##### Alpine ARM Cross-Building Requirements
+
+To cross-compile for _linux-musl_, most commonly for Alpine distros, you need to build the `binutils` from their repo, since unfortunately, they are not available as packages like the ones you installed in the encompassing section.
+
+To build them, follow these steps:
+
+* Clone the `musl-cross-make` repo (<https://github.com/richfelker/musl-cross-make>)
+* Create a new `config.mak` file in the root of the repo, and add the following lines into it:
+
+```makefile
+# Note that these two 'TARGET' clauses are for demonstration purposes only.
+# For an actual build, you have to only write one, since they are mutually exclusive.
+
+TARGET = armv6-alpine-linux-musleabihf # if cross-compiling for Arm32
+TARGET = aarch64-alpine-linux-musl     # if cross-compiling for Arm64
+OUTPUT = /usr
+BINUTILS_CONFIG=--enable-gold=yes
 ```
+
+* Run `make` from the root of the repo.
+* Run `sudo make install`, also from the root of the repo.
+
+### Unsupported OSes
```

Comment:
```suggestion
## Other Linux Distros
```
It is not clear what is unsupported about this.

Reply:
Agreed. Makes me wonder why it's been like that forever.